cocos/scene-graph/node.ts

```diff
@@ -2153,16 +2163,26 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
     public setRotation(x: number, y: number, z: number, w: number): void;
 
     public setRotation (val: Readonly<Quat> | number, y?: number, z?: number, w?: number): void {
+        let x: number;
+        const localRotation = this._lrot;
+
         if (y === undefined || z === undefined || w === undefined) {
```

Comment:
Should it be `and` logic here  like which in setPosition?

Reply:
I don't want to change its original logic in this PR.

Reply:
Okay.